### PR TITLE
Break up db_engine_specs

### LINKED ISSUE #7412: Break up db_engine_specs and add type hints
**Is your feature request related to a problem? Please describe.**
db_engine_specs has gotten large and should be broken up in multiple files. The class is also not completely typed. 

**Describe the solution you'd like**
There should be a file for each engine spec and the file should be typed.

### PR DESCRIPTION
**Pull request by villebro**
### CATEGORY

Choose one

- [x] Refactor

### SUMMARY
This PR changes `db_engine_specs.py` into a package with one module per engine, e.g. `db_engine_specs/bigquery.py`. All engines are automatically populated into `db_engine_specs.engines` like before (now in `db_engine_specs/__init__.py`), so only importing of specific engine specs will need to be changed:

Before: `from superset.db_engine_specs import BaseEngineSpec`
After: `from superset.db_engine_specs.base import BaseEngineSpec`

### TEST PLAN
Local testing + CI

### ADDITIONAL INFORMATION

- [x] Has associated issue: Closes #7412 

### REVIEWERS
@khtruong @john-bodley @betodealmeida @mistercrunch

### CODE REVIEW

File: superset/db_engine_specs/base.py [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+
+
+class LimitMethod(object):
+    """Enum the ways that limits can be applied"""
+    FETCH_MANY = 'fetch_many'
+    WRAP_SQL = 'wrap_sql'
+    FORCE_LIMIT = 'force_limit'
+
+
+def _create_time_grains_tuple(time_grains, time_grain_functions, blacklist):
```
Comment (john-bodley) [inline superset/db_engine_specs/base.py]: I would probably also nix the underscore here as it's a public method.

**Review by john-bodley — COMMENTED** [review]
Nice. I was thinking about making this same change. There's one small comment but overall LGTM.

**Comment by villebro** [comment]
Thanks for the quick comments @john-bodley, this should now be ready.

**Review by mistercrunch — APPROVED** [review]
LGTM - merge quick or conflict soon!

**Comment by villebro** [comment]
Pushing merge button in a few hours if no objections to avoid painful rebasing (already had 3 conflicts since initial commit).

**Comment by khtruong** [comment]
This is awesome. Thanks so much for this!

**Pull request merged**